source/extensions/filters/http/compressor/compressor_filter.cc

```diff
@@ -227,15 +227,13 @@ void CompressorFilter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallba
 Http::FilterHeadersStatus CompressorFilter::encodeHeaders(Http::ResponseHeaderMap& headers,
                                                           bool end_stream) {
   const auto& config = config_->responseDirectionConfig();
-  const bool isEnabledAndContentLengthBigEnough =
-      config.compressionEnabled() && config.isMinimumContentLength(headers);
-
   const bool isCompressible =
-      isEnabledAndContentLengthBigEnough && !Http::Utility::isUpgrade(headers) &&
-      config.isContentTypeAllowed(headers) && !hasCacheControlNoTransform(headers) &&
-      isEtagAllowed(headers) && !headers.getInline(response_content_encoding_handle.handle());
-  if (!end_stream && isEnabledAndContentLengthBigEnough && isAcceptEncodingAllowed(headers) &&
-      isCompressible && isTransferEncodingAllowed(headers)) {
+      config.compressionEnabled() && config.isMinimumContentLength(headers) &&
+      !Http::Utility::isUpgrade(headers) && config.isContentTypeAllowed(headers) &&
+      !hasCacheControlNoTransform(headers) && isEtagAllowed(headers) &&
+      !headers.getInline(response_content_encoding_handle.handle());
+  if (!end_stream && isAcceptEncodingAllowed(headers) && isCompressible &&
```

Comment:
Hi @rojkov 

Thanks for your review. In fact `isEnabledAndContentLengthBigEnough` is already part of `isCompressible` , so we don't need to evaluate this condition(`isEnabledAndContentLengthBigEnough`) in `if` . At the same time, since `isEnabledAndContentLengthBigEnough` is only used in B, A is merged into `isCompressible`.

Sorry I didn't make it clear before, I have added this part of the description to the pr description

Reply:
Like I said in #14115 `isAcceptEncodingAllowed()` isn't just a predicate. That's why I asked about unit tests.

I guess it would make sense to add a comment explaining why the condition looks this way to prevent future confusions.

Reply:
Hi @rojkov , Thanks for your reply. In fact all unit tests work well before, but I think I got your idea now. I have add some more test cases [here](https://github.com/envoyproxy/envoy/pull/20075/files#diff-5d1cb551420f188e04de06e22d8351dfa9b0fc375d4c4cc2671b87dfa99749f8R303). And do you mean in these new test cases no_accept_header should be `0`?

Could we make `isEnabledAndContentLengthBigEnough` as a parameter for `isAcceptEncodingAllowed`, and `isEnabledAndContentLengthBigEnough ` will be used inside `isAcceptEncodingAllowed`, just like below:
```suggestion
 if (!end_stream && isAcceptEncodingAllowed(isEnabledAndContentLengthBigEnough, headers) &&
      isCompressible && isTransferEncodingAllowed(headers))
```
and in isAcceptEncodingAllowed
```
bool CompressorFilter::isAcceptEncodingAllowed(const bool isEnabledAndContentLengthBigEnough, const Http::ResponseHeaderMap& headers) const {
  if (!isEnabledAndContentLengthBigEnough) {
    return false;
  }
  // origin logic
```

I think this will be more clear?

Reply:
Thank you! The tests look good to me. I'd like to have them. And yes, `no_accept_header` shouldn't be incremented in these tests: if compression is disabled we don't care about the header. The same is applicable to too small payloads. Also bear in mind that in case of too small payloads we shouldn't waste CPU cycles on calculating `EncodingDecision`.

Regarding extending the signature of `isAcceptEncodingAllowed()`... I'd leave it as it is for the sake of better code locality.